### PR TITLE
Add yamllint check and fix errors

### PR DESCRIPTION
**Pull request by piyush-garg**
This will enable the yamllint check in CI
for all yaml files except in vendor and
fix all the yamllint errors

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

### CODE REVIEW

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -29,10 +29,37 @@ export DISABLE_YAML_LINTING=1
...
+    fi
+
+    results_banner "Go Lint" 0
+}
+
+function check_yaml_lint() {
+    header "Testing if yamllint has been done"
+
+    local YAML_FILES=$(find . -path ./vendor -prune -o -type f -regex ".*y[a]ml" -print)
+    yamllint -c .yamllint ${YAML_FILES}
```
Comment (dlorenc) [inline test/presubmit-tests.sh:50]: Nice! This just happened to be installed in our image already?

Comment (piyush-garg) [reply test/presubmit-tests.sh:50]: Yes, yamllint binary was there already

**Pull request merged**